src/lib.rs

```diff
@@ -488,22 +497,46 @@ mod test {
     assert_eq!(data, xor(&xor(&data,&pad), &pad));
   }
 
+
+
   #[test]
-  fn check_write() {
+  fn check_write_and_read() {
```

Comment:
why are you combining a more simple test 'check_write' into a 'check_write_and_read' into a single test?